### PR TITLE
Add feature tracking

### LINKED ISSUE #230: Add usage tracking

### PR DESCRIPTION
**Pull request by gwdawson**
track executed queries so we can view some stats around the datasource
this includes but not limited to:
- how many queries are sent
- how often are each query types used
- ...

Fixes: [#72897](https://github.com/grafana/grafana/issues/72897)
Fixes #230

### CODE REVIEW

**Review by zoltanbedi — COMMENTED** [review]
This feels expensive. I'd rather use the dashboardLoaded event like we do in azure monitor https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/azuremonitor/module.ts#L24 so it won't create a new entry for every query run. WDYT?

**Comment by gwdawson** [comment]
based on what we discussed in todays standup, it seems we are happy to continue with my proposed solution.

File: src/DataSource.ts [hunk 1]
```diff
@@ -24,6 +26,49 @@ export class GithubDataSource extends DataSourceWithBackend<GitHubQuery, GithubD
...
+
+      const workflowTimeFieldReference: { [x: number]: string } = {
+        0: 'CreatedAt',
+        1: 'UpdatedAt',
+      };
+
+      let properties: Partial<GitHubQuery> = { app: request.app, queryType: target.queryType };
+
+      if (target.queryType === 'Issues') {
+        properties['timeField'] = issueTimeFieldReference[target.options?.timeField ?? 0];
```
Comment (zoltanbedi) [inline src/DataSource.ts:51]: I like how you don't just toss in the number here however this approach is not error prone. What if someone adds a new enum value to any of the types there? Your code will break.

Instead you can use https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings
like:
`IssueTimeField[target.options?.timeField ?? 0];`

Can you please also extract this reporting logic to its own function?
Any reason you are accessing the timeField property as an index signature?

**Review by zoltanbedi — CHANGES_REQUESTED** [review]

**Review by zoltanbedi — APPROVED** [review]
LGTM

**Pull request merged**